infrastructures/docker/infrastructure/src/main/java/org/eclipse/che/workspace/infrastructure/docker/provisioner/labels/SinglePortLabelsProvisioner.java

```diff
@@ -42,17 +42,21 @@
 
   @Inject
   public SinglePortLabelsProvisioner(
+      @Named("che.single.port") boolean isSinglePortEnabled,
       @Nullable @Named("che.docker.ip") String internalIpOfContainers,
       @Nullable @Named("che.docker.ip.external") String externalIpOfContainers,
       @Nullable @Named("che.docker.network") String dockerNetwork,
       @Nullable @Named("che.singleport.wildcard_domain.host") String wildcardHost) {
-    if (internalIpOfContainers == null && externalIpOfContainers == null) {
+    if (isSinglePortEnabled && internalIpOfContainers == null && externalIpOfContainers == null) {
       throw new IllegalStateException(
```

Comment:
I think `ConfigurationException` would be better here.

Reply:
Agree